### PR TITLE
chore: Remove old Crashlytics

### LINKED ISSUE #24: Release variant crashes on startup
**Describe the bug**
The app crashes on startup. The reason is the missing Crashlytics build. 
**To Reproduce**
1. Try opening the app

**Expected behavior**
The app should open and should display the main activity. 

**Observed behavior**
The app crashes when opened. 

**Device and Android version:** 

Google pixel 3 emulator. Android 10. Stock version. Google Play services version 20.04.14

**Stack trace**

> java.lang.RuntimeException: Unable to create application com.simplecity.amp_library.ShuttleApplication: io.a.a.a.a.c.m: The Crashlytics build ID is missing. This occurs when Crashlytics tooling is absent from your app's build configuration. Please review Crashlytics onboarding instructions and ensure you have a valid Crashlytics account.
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6465)
        at android.app.ActivityThread.access$1300(ActivityThread.java:219)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1859)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)

### PR DESCRIPTION
**Pull request by BumbleFlash**
Removed old crashlytics entirely

"Closes #24"

### CODE REVIEW

File: app/.gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 /build
+/keystore.jks
```
Comment (barbeau) [inline app/.gitignore]: Just curious - why does this need to be added to .gitignore?

File: app/build.gradle [hunk 2]
```diff
@@ -282,9 +283,9 @@ dependencies {
     compileOnly Dependencies.daggerAssistedInject
     kapt Dependencies.daggerAssistedInjectProcessor
 
-    implementation(Dependencies.crashlytics) {
-        transitive = true
-    }
+//    implementation(Dependencies.crashlytics) {
```
Comment (barbeau) [inline app/build.gradle]: Please remove entirely instead of commenting out.

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
```
Comment (barbeau) [inline app/src/main/AndroidManifest.xml]: Please try to keep unrelated changes out of the commit

File: app/src/main/java/com/simplecity/amp_library/playback/MusicService.java [hunk 4]
```diff
@@ -1014,7 +1014,6 @@ private void startForegroundImpl() {
                 Log.e(TAG, "startForeground should have been called, but song is null");
             }
         } catch (NullPointerException | ConcurrentModificationException e) {
-            Crashlytics.log("startForegroundImpl error: " + e.getMessage());
```
Comment (barbeau) [inline app/src/main/java/com/simplecity/amp_library/playback/MusicService.java:1017]: We shouldn't have any empty catch statements, as they are very dangerous. We should instead log these to `Log.e` - for example:

`Log.e(TAG, "startForegroundImpl error: " + e);`

TAG is a static final string with the name of the class defined in this class.

File: app/src/main/java/com/simplecity/amp_library/saf/SafManager.java [hunk 5]
```diff
@@ -228,7 +227,6 @@ private String getExtSdCardFolder(final File file) {
                 }
             }
         } catch (NoSuchMethodError e) {
-            Crashlytics.log("getExtSdCardPaths() failed. " + e.getMessage());
```
Comment (barbeau) [inline app/src/main/java/com/simplecity/amp_library/saf/SafManager.java:231]: Same here - no empty catch blocks

File: app/src/main/java/com/simplecity/amp_library/services/Equalizer.java [hunk 6]
```diff
@@ -308,7 +307,6 @@ public synchronized void update() {
                 updateDsp(mAudioSessions.get(sessionId));
             }
         } catch (NoSuchMethodError e) {
-            Crashlytics.log("No such method error thrown when updating equalizer.. " + e.getMessage());
```
Comment (barbeau) [inline app/src/main/java/com/simplecity/amp_library/services/Equalizer.java:311]: Same

File: app/src/main/java/com/simplecity/amp_library/ui/screens/tagger/TaggerUtils.java [hunk 7]
```diff
@@ -162,7 +161,6 @@ static boolean requiresPermission(Context context, List<String> paths) {
                 }
             }
         } catch (NoSuchMethodError e) {
-            Crashlytics.log("getExtSdCardPaths() failed. " + e.getMessage());
```
Comment (barbeau) [inline app/src/main/java/com/simplecity/amp_library/ui/screens/tagger/TaggerUtils.java:165]: Same

File: app/src/main/java/com/simplecity/amp_library/utils/playlists/PlaylistManager.kt [hunk 8]
```diff
@@ -200,7 +199,6 @@ class PlaylistManager @Inject constructor(
                             id = java.lang.Long.parseLong(uri.lastPathSegment!!)
                         }
                     } catch (e: NullPointerException) {
-                        Crashlytics.log("Failed to create playlist: " + e.message)
```
Comment (barbeau) [inline app/src/main/java/com/simplecity/amp_library/utils/playlists/PlaylistManager.kt:203]: Same

File: app/src/main/java/com/simplecity/amp_library/utils/playlists/PlaylistManager.kt [hunk 9]
```diff
@@ -210,7 +208,6 @@ class PlaylistManager @Inject constructor(
         if (id != -1L) {
             playlist = Playlist(Playlist.Type.USER_CREATED, id, name, true, false, true, true, true)
         } else {
-            Crashlytics.log(String.format("Failed to create playlist. Name: %s, id: %d", name, id))
```
Comment (barbeau) [inline app/src/main/java/com/simplecity/amp_library/utils/playlists/PlaylistManager.kt:213]: Same - log this via Log.e

File: app/src/main/java/com/simplecity/amp_library/utils/LogUtils.java [hunk 10]
```diff
@@ -18,8 +17,7 @@ public static void logException(String tag, String message, @Nullable Throwable
                 throwable.printStackTrace();
             }
         } else {
-            Crashlytics.log(Log.ERROR, tag, message + "\nThrowable: " + (throwable != null ? throwable.getMessage() : null));
```
Comment (barbeau) [inline app/src/main/java/com/simplecity/amp_library/utils/LogUtils.java:21]: Anywhere where Crashlytics was logging `Log.ERROR`, we should log to `Log.e`

File: build.gradle [hunk 11]
```diff
@@ -12,6 +12,7 @@ buildscript {
         classpath Dependencies.Plugins.gradleVersions
         classpath Dependencies.Plugins.playServices
         classpath Dependencies.Plugins.kotlin
+        classpath Dependencies.Plugins.firebase_crashlytics
```
Comment (barbeau) [inline build.gradle]: We can remove this, right?

File: buildSrc/src/main/kotlin/dependencies/Dependencies.kt [hunk 12]
```diff
@@ -87,6 +84,8 @@ object Dependencies {
             const val fabricGradlePlugin = "1.+"
             const val gradleVersions = "0.20.0"
             const val playServices = "4.2.0"
+            const val firebase_crashlytics="2.2.0"
```
Comment (barbeau) [inline buildSrc/src/main/kotlin/dependencies/Dependencies.kt]: We can remove this, right?

File: buildSrc/src/main/kotlin/dependencies/Dependencies.kt [hunk 13]
```diff
@@ -96,6 +95,8 @@ object Dependencies {
         const val playPublisher = "com.github.triplet.play"
         const val gradleVersions = "com.github.ben-manes:gradle-versions-plugin:${Versions.gradleVersions}"
         const val playServices = "com.google.gms:google-services:${Versions.playServices}"
+        const val firebase_crashlytics = "com.google.firebase:firebase-crashlytics-gradle:${Versions.firebase_crashlytics}"
```
Comment (barbeau) [inline buildSrc/src/main/kotlin/dependencies/Dependencies.kt]: We can remove this, right?

File: buildSrc/src/main/kotlin/dependencies/Dependencies.kt [hunk 14]
```diff
@@ -106,6 +107,8 @@ object Dependencies {
             const val firebaseRemoteConfig = "16.1.0"
             const val constraintLayout = "2.0.0-alpha3"
             const val chromeCastFramework = "16.1.0"
+
+            const val firebaseCrashlytics = "2.10.1"
```
Comment (barbeau) [inline buildSrc/src/main/kotlin/dependencies/Dependencies.kt]: We can remove this, right?

File: buildSrc/src/main/kotlin/dependencies/Dependencies.kt [hunk 15]
```diff
@@ -115,6 +118,7 @@ object Dependencies {
         const val supportv4 = "com.android.support:support-v4:${Versions.supportLib}"
         const val firebaseCore = "com.google.firebase:firebase-core:${Versions.firebaseCore}"
         const val firebaseRemoteConfig = "com.google.firebase:firebase-config:${Versions.firebaseRemoteConfig}"
+        const val firebaseCrashlytics = "com.google.firebase:firebase-crashlytics:${Versions.firebaseCrashlytics}"
```
Comment (barbeau) [inline buildSrc/src/main/kotlin/dependencies/Dependencies.kt]: We can remove this, right?

File: buildSrc/src/main/kotlin/dependencies/Dependencies.kt [hunk 16]
```diff
@@ -254,6 +258,7 @@ object Dependencies {
         const val fabric = "io.fabric"
         const val gradleVersions = "com.github.ben-manes.versions"
         const val playServices = "com.google.gms.google-services"
+        const val firebase_crashlytics = "com.google.firebase.crashlytics"
```
Comment (barbeau) [inline buildSrc/src/main/kotlin/dependencies/Dependencies.kt]: We can remove this, right?

**Review by barbeau — CHANGES_REQUESTED** [review]
Thanks @BumbleFlash! Some comments in-line - I think there are still some references to the new Crashlytics SDK, maybe from your experimentation, that need to be removed.

File: app/.gitignore (hunk 1 above)
Comment (BumbleFlash) [reply app/.gitignore]: I placed the keystore files in the project because I had trouble setting a relative path. But this is just a temporary one though!

File: app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (BumbleFlash) [reply app/src/main/AndroidManifest.xml]: Sure! This was me trying make the Firebase work without androidx. Sorry about that.

File: app/src/main/java/com/simplecity/amp_library/playback/MusicService.java (hunk 4 above)
Comment (BumbleFlash) [reply app/src/main/java/com/simplecity/amp_library/playback/MusicService.java:1017]: Ah, I thought I removed everything, Sorry about this!

Comment (BumbleFlash) [reply app/src/main/java/com/simplecity/amp_library/playback/MusicService.java:1017]: Got it will do that!

File: buildSrc/src/main/kotlin/dependencies/Dependencies.kt (hunk 12 above)
Comment (BumbleFlash) [reply buildSrc/src/main/kotlin/dependencies/Dependencies.kt:87]: Yes, we can.

File: buildSrc/src/main/kotlin/dependencies/Dependencies.kt (hunk 13 above)
Comment (BumbleFlash) [reply buildSrc/src/main/kotlin/dependencies/Dependencies.kt]: Yes, we can.

File: buildSrc/src/main/kotlin/dependencies/Dependencies.kt (hunk 14 above)
Comment (BumbleFlash) [reply buildSrc/src/main/kotlin/dependencies/Dependencies.kt]: Yes, we can.

File: buildSrc/src/main/kotlin/dependencies/Dependencies.kt (hunk 15 above)
Comment (BumbleFlash) [reply buildSrc/src/main/kotlin/dependencies/Dependencies.kt]: Yes, we can.

File: buildSrc/src/main/kotlin/dependencies/Dependencies.kt (hunk 16 above)
Comment (BumbleFlash) [reply buildSrc/src/main/kotlin/dependencies/Dependencies.kt]: Sure, we can. Again, curiosity getting the better of me. Sorry about that. :)

File: app/build.gradle (hunk 2 above)
Comment (BumbleFlash) [reply app/build.gradle]: Sure thing!

File: app/src/main/java/com/simplecity/amp_library/utils/LogUtils.java (hunk 10 above)
Comment (BumbleFlash) [reply app/src/main/java/com/simplecity/amp_library/utils/LogUtils.java:21]: Sorry about this. I was very curious about testing the app without crashlytics. Thanks for pointing these out.

File: build.gradle (hunk 11 above)
Comment (BumbleFlash) [reply build.gradle]: Sure, we can.

File: buildSrc/src/main/kotlin/dependencies/Dependencies.kt (hunk 12 above)
Comment (BumbleFlash) [reply buildSrc/src/main/kotlin/dependencies/Dependencies.kt]: Sure, we can.

File: app/.gitignore (hunk 1 above)
Comment (barbeau) [reply app/.gitignore]: Can you please remove from the PR if this is temporary?

File: app/src/main/java/com/simplecity/amp_library/utils/ThreadUtils.java [hunk 17]
```diff
@@ -15,7 +14,6 @@ public static void ensureNotOnMainThread() {
             if (BuildConfig.DEBUG) {
                 throw new IllegalStateException("ensureNotOnMainThread failed.");
             } else {
-                CrashlyticsCore.getInstance().log("ThreadUtils ensureNotOnMainThread() failed");
```
Comment (barbeau) [inline app/src/main/java/com/simplecity/amp_library/utils/ThreadUtils.java:18]: This shouldn't be an empty else statement - either remove else block or log this as an error like the others.

File: app/build.gradle [hunk 18]
```diff
@@ -226,6 +225,7 @@ dependencies {
     implementation Dependencies.Google.supportv4
     implementation Dependencies.Google.firebaseCore
     implementation Dependencies.Google.firebaseRemoteConfig
+
```
Comment (barbeau) [inline app/build.gradle]: Please try to remove any extra spaces or newlines.

File: app/src/main/java/com/simplecity/amp_library/services/Equalizer.java [hunk 19]
```diff
@@ -308,8 +307,8 @@ public synchronized void update() {
...
-        }
+            Log.e(TAG, "No such method error thrown when updating equalizer.. " + e);
+    }
```
Comment (barbeau) [inline app/src/main/java/com/simplecity/amp_library/services/Equalizer.java]: Looks like this `}` should be indented.

File: app/.gitignore (hunk 1 above)
Comment (BumbleFlash) [reply app/.gitignore]: Sure thing. Do you have a specific path for the `keystore.jks` that you'd suggest? I can include the path in the `keystore.properties` file.

Comment (barbeau) [reply app/.gitignore]: Path to `keystore.jks` should be somewhere unversioned on your own computer outside of the project folder. `keystore.properties` file goes in the root project folder (you can see it's currently in the `.gitignore` there, so it's not versioned either), and then you can put the absolute path to the `keystore.jks` file in that file, like:

```
store=D:/Dropbox/MUSER/keystore/keystore.jks
...
```

Comment (BumbleFlash) [reply app/.gitignore]: Great. Thank you for letting me know.

**Review by barbeau — APPROVED** [review]
